packages/pds/src/app-view/services/indexing/plugins/post.ts

```diff
@@ -76,34 +77,47 @@ const insertFn = async (
     return []
   })
   // Embed indices
-  let embed: PostEmbedImage[] | PostEmbedExternal | PostEmbedRecord | undefined
-  if (isEmbedImage(obj.embed)) {
-    const { images } = obj.embed
-    embed = images.map((img, i) => ({
-      postUri: uri.toString(),
-      position: i,
-      imageCid: img.image.ref.toString(),
-      alt: img.alt,
-    }))
-    await db.insertInto('post_embed_image').values(embed).execute()
-  } else if (isEmbedExternal(obj.embed)) {
-    const { external } = obj.embed
-    embed = {
-      postUri: uri.toString(),
-      uri: external.uri,
-      title: external.title,
-      description: external.description,
-      thumbCid: external.thumb?.ref.toString() || null,
-    }
-    await db.insertInto('post_embed_external').values(embed).execute()
-  } else if (isEmbedRecord(obj.embed)) {
-    const { record } = obj.embed
-    embed = {
-      postUri: uri.toString(),
-      embedUri: record.uri,
-      embedCid: record.cid,
+  const embeds: (PostEmbedImage[] | PostEmbedExternal | PostEmbedRecord)[] = []
+  const postEmbeds = isEmbedComplexRecord(obj.embed)
+    ? [
+        { $type: lex.ids.AppBskyEmbedRecord, ...obj.embed.record },
+        obj.embed.media,
+      ]
+    : obj.embed
```

Comment:
can we make this an explicit check (`obj.embed !== undefined`)? just for readabilities sake? thought it was a value at first